pkgs/servers/mpd/default.nix

```diff
@@ -30,13 +30,17 @@
 
 let
   lib = stdenv.lib;
+  concatAttrVals = nameList: set: lib.concatMap (x: set.${x} or []) nameList;
 
   featureDependencies = {
     # Storage plugins
     udisks        = [ dbus ];
     webdav        = [ curl expat ];
     # Input plugins
     curl          = [ curl ];
+    # Disabled because the mpd binary has the wrong output of liburing in its
+    # RUNPATH
```

Comment:
Could you elaborate please?

Reply:
If `io_uring` is enabled, the build itself succeeds, but the tests fail because the binary isn't built correctly. `liburing.so` isn't found because the wrong output of `liburing` is inserted into the `RUNPATH` of the `mpd` binary. The meson wrapper of `pkg-config` seems to be unable to deal with the way `liburing` is installed in nixpkgs. If I remove the `lib` output of `liburing`, then everything builds, but I don't want to do that in this PR because it would trigger a much larger rebuild.

Reply:
> f io_uring is enabled, the build itself succeeds, but the tests fail because the binary isn't built correctly. liburing.so isn't found because the wrong output of liburing is inserted into the RUNPATH of the mpd binary. The meson wrapper of pkg-config seems to be unable to deal with the way liburing is installed in nixpkgs. If I remove the lib output of liburing, then everything builds.

That's because `liburing` is not built properly (cc @thoughtpolice):

```
$ tree $(nix-build --no-out-link -A liburing.out)
/nix/store/vvjjmlhnfph4ah8zv2dzyc6i7ccp6404-liburing-0.7
├── bin
│   ├── io_uring-cp
│   ├── io_uring-link-cp
│   ├── io_uring-test
│   └── io_uring-ucontext-cp
└── lib
    ├── liburing.a
    └── liburing.so -> /nix/store/pi16ynlm7fyvxxsq7xrd3s2k40ryn9ss-liburing-0.7-lib/lib/liburing.so.1.0.7

2 directories, 6 files
```

`liburing.so` is referenced between the "lib" and "out" outputs, that's why disabling "lib" worked for you. This is an issue with liburing and not mpd. 

> because it would trigger a much larger rebuild.

Fortunately, `liburing` is used only by mpd and samba, which luckily both have NixOS tests we can use :) 

I'm pushing a change to liburing, for which I tested samba to at least build with it, and I checked that liburing support was included:

```
$ nix-store --read-log $(nix-build --no-out-link -A samba) | grep uring
configuring
Checking for liburing package                                                                   : yes 
Checking for header liburing.h                                                                  : yes 
Checking for library uring                                                                      : yes 
Checking for io_uring_ring_dontfork in uring                                                    : ok 
VFS_SHARED: vfs_recycle,vfs_audit,vfs_extd_audit,vfs_full_audit,vfs_fake_perms,vfs_default_quota,vfs_readonly,vfs_cap,vfs_expand_msdfs,vfs_shadow_copy,vfs_shadow_copy2,vfs_readahead,vfs_xattr_tdb,vfs_streams_xattr,vfs_streams_depot,vfs_acl_xattr,vfs_acl_tdb,vfs_preopen,vfs_catia,vfs_media_harmony,vfs_unityed_media,vfs_fruit,vfs_shell_snap,vfs_commit,vfs_worm,vfs_crossrename,vfs_linux_xfs_sgid,vfs_time_audit,vfs_offline,vfs_virusfilter,vfs_syncops,vfs_dirsort,vfs_fileid,vfs_aio_fork,vfs_io_uring,vfs_aio_pthread,vfs_gpfs,vfs_btrfs,vfs_glusterfs_fuse
[3167/3704] Compiling source3/modules/vfs_io_uring.c
[3179/3704] Linking bin/default/source3/modules/libvfs_module_io_uring.so
[3765/3818] Compiling docs-xml/manpages/vfs_io_uring.8.xml
docs-xml/manpages/vfs_io_uring.8.xml:2: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd"
Note: Writing vfs_io_uring.8
[4004/4735] Linking bin/default/source3/modules/libvfs_module_io_uring.inst.so
[4182/4735] Linking bin/default/source3/modules/libvfs_module_io_uring.so
+ install /nix/store/lzz0jrwabnxfj9j2dmwgnv5mgznhmr81-samba-4.12.5/lib/samba/vfs/io_uring.so (from bin/default/source3/modules/libvfs_module_io_uring.inst.so)
+ install /nix/store/0jnicddaxyrbg3mdicgpdygir8bxp5fb-samba-4.12.5-man/share/man/man8/vfs_io_uring.8 (from bin/default/docs-xml/manpages/vfs_io_uring.8)
shrinking /nix/store/lzz0jrwabnxfj9j2dmwgnv5mgznhmr81-samba-4.12.5/lib/samba/vfs/io_uring.so
```